### PR TITLE
App/Activity: Use ViewModel and Jetpack Compose UI

### PR DESCRIPTION
**Pull request by niley7464**
This PR uses ViewModel and Jetpack Compose for UI development.
ViewModel uses dependency injection.
Also button's onClick method is changed to use AppDatabase.

### CODE REVIEW

File: ml_inference_offloading/src/main/java/ai/nnstreamer/ml/inference/offloading/MainService.kt [hunk 1]
```diff
@@ -87,8 +152,13 @@ class MainService : Service() {
     @Inject
     lateinit var modelsRepository: ModelRepositoryImpl
 
+    @Inject
+    lateinit var offloadingServiceRepositoryImpl: OffloadingServiceRepositoryImpl
+
     private var initialized = false
     private var serverInfoMap = mutableMapOf<String, ServerInfo>()
+    private var serviceMap = mutableMapOf<Int, OffloadingServiceStatus>()
+    private var gServiceId = 1
```
Comment (wooksong) [inline ml_inference_offloading/src/main/java/ai/nnstreamer/ml/inference/offloading/MainService.kt:161]: If you want to use a persistent variable that is initialized only once when the app is installed, it should be moved to the room or datastore. I think this variable could be initialized every time the service restarted.

File: ml_inference_offloading/src/main/java/ai/nnstreamer/ml/inference/offloading/MainService.kt [hunk 2]
```diff
@@ -261,65 +311,76 @@ class MainService : Service() {
...
-                    Log.e(TAG, "unregistration failed")
+            it.forEach { model ->
+                val port = findPort()
+                // TODO: This is a temporary desc and must be updated to use model info correctly
+                val desc =
+                    "tensor_query_serversrc host=" + hostAddress + " port=" +
+                            port.toString() + " ! " + model.name + " ! tensor_query_serversink async=false"
+
+                CoroutineScope(Dispatchers.IO).launch {
+                    // TODO: Find a better way to create new uid
```
Comment (wooksong) [inline ml_inference_offloading/src/main/java/ai/nnstreamer/ml/inference/offloading/MainService.kt:345]: Got it.

**Comment by niley7464** [comment]
ANR occurs frequently and needs to be fixed
<details>
<summary>log</summary>

```
ANR in ai.nnstreamer.ml.inference.offloading (ai.nnstreamer.ml.inference.offloading/.MainActivity)
PID: 27137
Reason: Input dispatching timed out (256520a ai.nnstreamer.ml.inference.offloading/ai.nnstreamer.ml.inference.offloading.MainActivity (server) is not responding. Waited 10004ms for MotionEvent(deviceId=4, eventTime=766625543204000, source=TOUCHSCREEN, displayId=0, action=UP, actionButton=0x00000000, flags=0x00000000, metaState=0x00000000, buttonState=0x00000000, classification=NONE, edgeFlags=0x00000000, xPrecision=3.8, yPrecision=1.8, xCursorPosition=nan, yCursorPosition=nan, pointers=[0: (126.3, 1547.1)]), policyFlags=0x62000800)
Parent: ai.nnstreamer.ml.inference.offloading/.MainActivity
ErrorId: a4529fe9-5436-4961-9e91-2293b209ae8b
Frozen: s[false] g[false]
Load: 14.69 / 14.3 / 14.16
------ Current CPU Core Info ------
- offline : 
- online : 0-9
- AP Temp = 301
                           0           1           2           3           4           5           6           7           8           9
------------------------------------------------------------------------------------------------------------------------------------------
scaling_cur_freq       1152000     1152000     1152000     1152000      672000      672000      672000      672000      672000      672000
scaling_governor  energy_awareenergy_awareenergy_awareenergy_awareenergy_awareenergy_awareenergy_awareenergy_awareenergy_awareenergy_aware
scaling_max_freq       1959000     1959000     1959000     1959000     2592000     2592000     2592000     2900000     2900000     3207000
------------------------------------------------------------------------------------------------------------------------------------------
----- Output from /proc/pressure/memory -----
some avg10=0.00 avg60=0.00 avg300=0.00 total=68750111
full avg10=0.00 avg60=0.00 avg300=0.00 total=52458205
----- End output from /proc/pressure/memory -----
----- Output from /proc/pressure/cpu -----
some avg10=5.21 avg60=4.60 avg300=2.89 total=17046208511
full avg10=0.00 avg60=0.00 avg300=0.00 total=0
----- End output from /proc/pressure/cpu -----
----- Output from /proc/pressure/io -----
some avg10=0.00 avg60=0.00 avg300=0.00 total=181647176
full avg10=0.00 avg60=0.00 avg300=0.00 total=132418268
----- End output from /proc/pressure/io -----

CPU usage from 100455ms to -1ms ago (2024-08-01 10:56:24.831 to 2024-08-01 10:58:05.287):
9.1% 1385/system_server: 4.7% user + 4.3% kernel / faults: 3460 minor 1 major
4.6% 8734/adbd: 1.5% user + 3% kernel / faults: 2664 minor
3.5% 10807/com.google.android.gms.persistent: 2.4% user + 1% kernel / faults: 3478 minor 31 major
2.5% 464/logd: 0.4% user + 2% kernel / faults: 580 minor
2.4% 1195/surfaceflinger: 1.2% user + 1.2% kernel / faults: 47 minor
1.8% 2986/com.android.bluetooth: 1.1% user + 0.7% kernel / faults: 619 minor 1082 major
1.7% 1344/cnss_diag: 1.3% user + 0.4% kernel
1.7% 23805/logcat: 0.3% user + 1.4% kernel
1.5% 150/pagov: 0% user + 1.5% kernel
1.4% 8826/process-tracker: 0.2% user + 1.1% kernel
6.4% TOTAL: 1.9% user + 3.1% kernel + 0% iowait + 1% irq + 0.2% softirq
CPU usage from 608ms to 1074ms later (2024-08-01 10:58:05.894 to 2024-08-01 10:58:06.360):
64% 1385/system_server: 24% user + 40% kernel / faults: 481 minor 3 major
  40% 1414/Signal Catcher: 24% user + 16% kernel
  16% 27648/AnrAuxiliaryTas: 0% user + 16% kernel
21% 4940/com.sec.android.sdhms: 4.3% user + 17% kernel / faults: 393 minor 799 major
  17% 5023/Signal Catcher: 8.6% user + 8.6% kernel
  4.3% 5029/HeapTaskDaemon: 4.3% user + 0% kernel
21% 7000/com.samsung.android.honeyboard: 8.7% user + 13% kernel / faults: 656 minor 44 major
  17% 7021/Signal Catcher: 8.7% user + 8.7% kernel
  4.3% 7162/Profile Saver: 4.3% user + 0% kernel
8.6% 4914/com.samsung.android.sead: 8.6% user + 0% kernel / faults: 371 minor 958 major
  12% 4999/Signal Catcher: 8.6% user + 4.3% kernel
8.8% 8734/adbd: 4.4% user + 4.4% kernel
  4.4% 8734/adbd: 4.4% user + 0% kernel
```
</details>

File: ml_inference_offloading/src/main/java/ai/nnstreamer/ml/inference/offloading/MainService.kt (hunk 1 above)
Comment (niley7464) [reply ml_inference_offloading/src/main/java/ai/nnstreamer/ml/inference/offloading/MainService.kt:161]: I am considering using the `PrimaryKey(autoGenerate = true)` option.

**Review by wooksong — APPROVED** [review]
Looks good!

**Pull request merged**